SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -349,16 +353,53 @@ SPATIALGDKEDITOR_API void GenerateSchemaForSublevels(const FString& SchemaOutput
 	Writer.WriteToFile(FString::Printf(TEXT("%sSublevels/sublevels.schema"), *SchemaOutputPath));
 }
 
-SPATIALGDKEDITOR_API void GenerateSchemaForRPCEndpoints()
+void GenerateSchemaForRPCEndpoints()
 {
 	GenerateSchemaForRPCEndpoints(GetDefault<USpatialGDKEditorSettings>()->GetGeneratedSchemaOutputFolder());
 }
 
-SPATIALGDKEDITOR_API void GenerateSchemaForRPCEndpoints(const FString& SchemaOutputPath)
+void GenerateSchemaForRPCEndpoints(const FString& SchemaOutputPath)
 {
 	GenerateRPCEndpointsSchema(SchemaOutputPath);
 }
 
+void GenerateSchemaForNCDs()
+{
+	GenerateSchemaForNCDs(GetDefault<USpatialGDKEditorSettings>()->GetGeneratedSchemaOutputFolder());
+}
+
+void GenerateSchemaForNCDs(const FString& SchemaOutputPath)
+{
+	FCodeWriter Writer;
+	Writer.Printf(R"""(
+		// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+		// Note that this file has been generated automatically
+		package unreal.ncdcomponents;)""");
+
+	FComponentIdGenerator IdGenerator = FComponentIdGenerator(NextAvailableComponentId);
+
+	for (auto& NCDComponent : NetCullDistanceToComponentId)
+	{
+		const FString ComponentName = FString::Printf(TEXT("NetCullDistanceSquared%u"), static_cast<uint32>(NCDComponent.Key));
```

Comment:
Can we add a warning or something for when a user is trying to shoot themselves with fractional values?